wire-ios-data-model/Source/Model/UserClient/UserClientType.swift

```diff
@@ -37,6 +37,8 @@ import Foundation
     /// True if the self user has verfied the client
     var verified: Bool { get }
 
+    var needsToNotifyUser: Bool { get set }
+
```

Comment:
question: I don't see the variable `needsToNotifyUser` used in the changes, why was it added?

Reply:
Good question! When I changed the type from UserClient to UserClientType I would get an error in this method:

```swift
fileprivate func presentNewLoginAlertController(_ clients: [UserClientType]) {
        let newLoginAlertController = UIAlertController(forNewSelfClients: clients)

        let actionManageDevices = UIAlertAction(title: L10n.Localizable.Self.NewDeviceAlert.manageDevices, style: .default) { _ in
            self.openControllerForCellWithIdentifier(SettingsCellDescriptorFactory.settingsDevicesCellIdentifier)
        }

        newLoginAlertController.addAction(actionManageDevices)

        let actionTrustDevices = UIAlertAction(title: L10n.Localizable.Self.NewDeviceAlert.trustDevices, style: .default) { [weak self] _ in
            self?.presentUserSettingChangeControllerIfNeeded()
        }

        newLoginAlertController.addAction(actionTrustDevices)

        present(newLoginAlertController, animated: true, completion: .none)

        userSession.enqueue {
            clients.forEach {
                $0.needsToNotifyUser = false
            }
        }
    }
    ```

More specifically in line:

**$0.needsToNotifyUser = false**

needsToNotifyUser wasn't part of UserClientType and I needed to add it.

Reply:
thanks!